File: src/test/java/dirtymaster/fgpt/client/FreedomGPTTemplateTest.java
package dirtymaster.fgpt.client;

import dirtymaster.fgpt.model.Model;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertFalse;

class FreedomGPTTemplateTest {
    private final String apiKey;

    FreedomGPTTemplateTest() throws IOException {
        this.apiKey = new String(Files.readAllBytes(Paths.get("src/test/resources/key.txt")));
    }

    @Test
    void testCompletion() {
        FreedomGPTTemplate template = new FreedomGPTTemplate(Model.CLAUDE_3_5_SONNET, apiKey);
        String hello = template.getCompletion("hello");
        assertFalse(hello.isEmpty());
    }
}

File: src/test/java/dirtymaster/fgpt/client/FreedomGPTTemplateTest.kt
package dirtymaster.fgpt.client

import dirtymaster.fgpt.model.Model
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.nio.file.Files
import java.nio.file.Paths

internal class FreedomGPTTemplateTest {
    private val apiKey = String(Files.readAllBytes(Paths.get("src/test/resources/key.txt")))

    @Test
    fun testCompletion() {
        val template = FreedomGPTTemplate(Model.claudeMinus3Period5MinusSonnet, apiKey)
        val hello = template.getCompletion("hello")
        Assertions.assertFalse(hello.isEmpty())
    }
}